File: android/app/src/main/java/com/scascan/app/domain/usecase/AnalyzeFoodImageUseCase.kt
package com.scascan.app.domain.usecase

import android.graphics.Bitmap
import com.scascan.app.data.model.NutritionFacts
import com.scascan.app.data.repository.NutritionRepository
import javax.inject.Inject

class AnalyzeFoodImageUseCase @Inject constructor(
    private val repository: NutritionRepository
) {
    suspend operator fun invoke(bitmap: Bitmap): Result<NutritionFacts> =
        repository.analyzeImage(bitmap)
}

File: ios/ScaScanKit/Sources/ScaScanKit/Domain/UseCases/AnalyzeFoodImageUseCase.swift
import UIKit

/// Mirrors Android's `domain.usecase.AnalyzeFoodImageUseCase`.
public struct AnalyzeFoodImageUseCase: Sendable {
    private let repository: NutritionRepository

    public init(repository: NutritionRepository) {
        self.repository = repository
    }

    public func callAsFunction(_ image: UIImage) async throws -> NutritionFacts {
        try await repository.analyzeImage(image)
    }
}
